public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -79,17 +79,22 @@ export default class CloudMonitoringDatasource extends DataSourceWithBackend<
       })
       .pipe(
         map(({ data }) => {
-          const results = data.results['annotationQuery'].tables[0].rows.map((v: any) => {
-            return {
-              annotation: annotation,
-              time: Date.parse(v[0]),
-              title: v[1],
-              tags: [],
-              text: v[3],
-            } as any;
+          const dataQueryResponse = toDataQueryResponse({
+            data: data,
           });
-
-          return results;
+          const df: any = [];
+          if (dataQueryResponse.data.length !== 0) {
+            for (let i = 0, j = 0; i < dataQueryResponse.data.length, j < dataQueryResponse.data[i].length; i++, j++) {
+              df.push({
```

Comment:
If the result from the parseToAnnotation method was returned from the backend, there would be no need to do this mapping in the frontend

Reply:
Discussed this, I think it's the only way to iterate through dataframes :)

Also split for loop into two different.